ethereum/contracts/relayer/coreRelayer/CoreRelayerSerde.sol

```diff
@@ -0,0 +1,272 @@
+// SPDX-License-Identifier: Apache 2
+
+pragma solidity ^0.8.19;
+
+import {
+  InvalidPayloadId,
+  InvalidPayloadLength,
+  InvalidVaaKeyType,
+  VaaKey,
+  VaaKeyType,
+  Send,
+  ExecutionParameters,
+  DeliveryInstruction,
+  RedeliveryInstruction,
+  DeliveryOverride
+} from "../../interfaces/relayer/IWormholeRelayer.sol";
+import {BytesParsing} from "./BytesParsing.sol";
+
+library CoreRelayerSerde {
+  using BytesParsing for bytes;
+
+  // ---------------------- "public" (i.e implicitly internal) encode/decode -----------------------
+
+  uint8 private constant PAYLOAD_ID_VAAKEY = 1;
+  uint8 private constant PAYLOAD_ID_EXECUTION_PARAMETERS = 1;
+  uint8 private constant PAYLOAD_ID_DELIVERY_INSTRUCTION = 1;
+  uint8 private constant PAYLOAD_ID_DELIVERY_OVERRIDE = 1;
+  //TODO AMO: Why are we at version 2 here?
```

Comment:
When reading wormhole messages emitted from the contract (in the relayer engine), we parse the first byte to determine whether it is a DeliveryInstruction or a RedeliveryInstruction.